Automattic/jetpack
# Licensing: Add support for Jetpack licenses

~Blocked by #17009~
~Requires D48540-code~

#### Changes proposed in this Pull Request:
* Automatically attach licenses stored in the `jetpack_licenses` option whenever Jetpack is connected or the option value is updated (i.e. provision the products associated with the licenses for the current site).
* Surface an error notice in the Jetpack dashboard whenever a license attach request fails. 

#### Jetpack product discussion
pbtFPC-EA-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
Case 1:
- Make sure no plans/products are provisioned for the current site.
- Store one or more (valid and invalid) license keys in an array as the value for the `jetpack_licenses` option.
- Connect (or reconnect) Jetpack.
- The plans/products associated with the valid licenses should now be provisioned. The presence of invalid/revoked license keys should result in an error notice listing the offending license keys.

Case 2:
- Same as Case 1 but instead of reconnecting Jetpack keep it connected and update the value of `jetpack_licenses` to make sure changes trigger a revalidation of stored licenses.

#### Proposed changelog entry for your changes:
No changelog entry needed for now.
